dotnet/ClientLib/MemoryWebClient.cs

```diff
@@ -191,26 +192,29 @@ public async Task DeleteDocumentAsync(string documentId, string? index = null, C
         return status;
     }
 
-    /// <inheritdoc />
-    public Task<SearchResult> SearchAsync(
-        string query,
-        MemoryFilter? filter,
-        int limit = -1,
-        CancellationToken cancellationToken = default)
-    {
-        return this.SearchAsync(query: query, index: null, filter, limit, cancellationToken);
-    }
-
     /// <inheritdoc />
     public async Task<SearchResult> SearchAsync(
         string query,
         string? index = null,
         MemoryFilter? filter = null,
+        ICollection<MemoryFilter>? filters = null,
         int limit = -1,
         CancellationToken cancellationToken = default)
     {
+        if (filter != null)
+        {
+            if (filters == null) { filters = new List<MemoryFilter>(); }
+
+            filters.Add(filter);
+        }
+
+        if (filters is { Count: > 1 })
+        {
+            throw new ArgumentException("The method does not support a filter list greater than 1", nameof(filters));
```

Comment:
any reason for this exception? what's missing in order to support multiple filters?

Reply:
Think this is outdated per my first pass on this. Will review and modify appropriately in a bit.

Reply:
Removed as there shouldn't be a limitation here. Think I absent mindedly added as a fail-fast thinking the underlying provider could have been QDrant.

Reply:
same in `AskAsync` right? 

I'll take care of that one because it's breaking some functional tests